pandas/tseries/index.py

```diff
@@ -1447,7 +1447,11 @@ def _maybe_cast_slice_bound(self, label, side, kind):
                            getattr(self, 'inferred_freq', None))
             _, parsed, reso = parse_time_string(label, freq)
             bounds = self._parsed_string_to_bounds(reso, parsed)
-            return bounds[0 if side == 'left' else 1]
+            bounds_side = 0 if side == 'left' else 1
```

Comment:
@shoyer - I ended up writing this a bit differently, unpacking `bounds` into 2 names.  Let me know if you think it could be any clearer.